src/chronify/time_series_mapper_representative.py

```diff
@@ -0,0 +1,176 @@
+import time
+
+import pandas as pd
+
+from chronify.sqlalchemy.functions import read_database, write_database
+from chronify.models import TableSchema
+from chronify.exceptions import MissingParameter, ConflictingInputsError
+from chronify.time_series_mapper_base import TimeSeriesMapperBase
+from chronify.time import (
+    representative_weekday_column,
+)
+from chronify.utils.sqlalchemy_view import create_view
+
+
+class MapperRepresentativeTimeToDatetime(TimeSeriesMapperBase):
+    def __init__(
+        self, engine: Engine, metadata: MetaData, from_schema: TableSchema, to_schema: TableSchema
+    ) -> None:
+        self._engine = engine
+        self._metadata = metadata
+        self.from_schema = from_schema
+        self.to_schema = to_schema
+        self.weekday_column = representative_weekday_column[from_schema.time_config.time_format]
+
+    def check_schema_consistency(self) -> None:
+        self._check_table_column_producibility()
+        self._check_schema_measurement_type_consistency()
+
+    def _check_table_column_producibility(self) -> None:
+        available_cols = self.from_schema.list_columns() + [self.to_schema.time_config.time_column]
+        final_cols = self.to_schema.list_columns()
+        if diff := set(final_cols) - set(available_cols):
+            msg = f"Source table {self.from_schema.time_config.name} cannot produce the columns: {diff}"
+            raise ConflictingInputsError(msg)
+
+    def _check_schema_measurement_type_consistency(self) -> None:
+        from_mt = self.from_schema.time_config.measurement_type
+        to_mt = self.from_schema.time_config.measurement_type
+        if from_mt != to_mt:
+            msg = f"Inconsistent measurement_types {from_mt=} vs. {to_mt=}"
+            raise ConflictingInputsError(msg)
+
+    def _check_source_table_has_time_zone(self) -> None:
+        if "time_zone" not in self.from_schema.time_array_id_columns:
+            msg = f"time_zone is required for representative time mapping and it is missing from source table: {self.from_schema.time_config.name}"
+            raise MissingParameter(msg)
+
+    def map_time(self) -> None:
+        # Destination time
+        to_time_col = self.to_schema.time_config.time_column
+        dft = (
+            pd.Series(self.to_schema.time_config.list_timestamps()).rename(to_time_col).to_frame()
+        )
+
+        # Apply checks
+        self.check_schema_consistency()
+        if not self.to_schema.time_config.is_time_zone_naive():
+            self._check_source_table_has_time_zone()
+        # TODO: add interval type handling (note annual has no interval type)
+
+        # Create mapping
+        if self.to_schema.time_config.is_time_zone_naive():
+            dfm = self._create_mapping_dataframe_tz_naive(dft, to_time_col)
+        else:
+            dfm = self._create_mapping_dataframe_tz_aware(dft, to_time_col)
+
+        # Ingest mapping into db
+        time_array_id_columns = [
```

Comment:
This is a workaround, because the mapping table does not have a value column. The table schema is not super important here but is needed to use the `write_database` func.